engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStoreBenchmark.scala

```diff
@@ -40,30 +40,36 @@ object ExecutionStoreBenchmark extends Bench[Double] with DefaultJsonProtocol {
   private def makeKey(call: CommandCallNode, executionStatus: ExecutionStatus)(index: Int) = {
     BackendJobDescriptorKey(call, Option(index), 1) -> executionStatus
   }
-  
-  // Generates numbers from 1000 to 10000 with 1000 gap:
-  // 1000, 2000, ..., 10000
-  val sizes: Gen[Int] = Gen.range("size")(1000, 10000, 1000)
-  
+
   // Generates executionStores using the given above sizes
   // Each execution store contains X simulated shards of "prepareCall" in status Done and X simulated shards of "scatterCall" in status NotStarted
   // This provides a good starting point to evaluate the speed of "runnableCalls", as it needs to iterate over all "NotStarted" keys, and for each one
   // look for their upstreams keys in status "Done"
-  val executionStores: Gen[ActiveExecutionStore] = for {
-    size <- sizes
-    doneMap = (0 until size map makeKey(prepareCall, ExecutionStatus.Done)).toMap
-    collectorKeys = scatter.outputMapping.map(om => ScatterCollectorKey(om, size, ScatterNode.DefaultScatterCollectionFunction) -> ExecutionStatus.NotStarted).toMap
-    notStartedMap = (0 until size map makeKey(scatterCall, ExecutionStatus.NotStarted)).toMap ++ collectorKeys
-    finalMap: Map[JobKey, ExecutionStatus] = doneMap ++ notStartedMap
-  } yield ActiveExecutionStore(finalMap, true)
+  private def stores(sizes: Gen[Int]): Gen[ActiveExecutionStore] = {
+    for {
+      size <- sizes
+      doneMap = (0 until size map makeKey(prepareCall, ExecutionStatus.Done)).toMap
+      collectorKeys = scatter.outputMapping.map(om => ScatterCollectorKey(om, size, ScatterNode.DefaultScatterCollectionFunction) -> ExecutionStatus.NotStarted).toMap
+      notStartedMap = (0 until size map makeKey(scatterCall, ExecutionStatus.NotStarted)).toMap ++ collectorKeys
+      finalMap: Map[JobKey, ExecutionStatus] = doneMap ++ notStartedMap
+    } yield ActiveExecutionStore(finalMap, needsUpdate = true)
+  }
   
   performance of "ExecutionStore" in {
     // Measures how fast the execution store can find runnable calls with lots of "Done" calls and "NotStarted" calls.
     // Other "shapes" would be valuable to get a better sense of how this method behaves in various situations (with Collector Keys etc...)  
-    measure method "runnableCalls" in {
-      using(executionStores) in { es =>
+    measure method "update" in {
+      val sizes: Gen[Int] = Gen.range("size")(from = 1000, upto = 10000, hop = 1000)
+      using(stores(sizes)) in { es =>
         es.update
       }
     }
+
+    measure method "backendJobDescriptorKeyForNode" in {
+      val sizes: Gen[Int] = Gen.range("size")(from = 100000, upto = 1000000, hop = 100000)
+      using(stores(sizes)) in { es =>
+        es.backendJobDescriptorKeyForNode(scatterCall)
```

Comment:
Hmm, I wonder whether the test is not quite representative. If _all_ of the keys in the list are scatters and none are `BackendJobDescriptorKey ` then we presumably won't be exercising the equality check performance?

Reply:
Discussed with faces, the existing test was a bit confusing. All the keys are actually `BackendJobDescriptorKey`s.